indicnlp/tokenize/sentence_tokenize.py

```diff
@@ -32,6 +31,19 @@
 ## pattern to check for presence of danda in text 
 CONTAINS_DANDA=re.compile(r'[\u0964\u0965]')
 
+def is_latin_or_numeric(character):
+    """
+    Check if a character is a Latin character (uppercase or lowercase) or a number.
+
+    Parameters:
+        character (str): The character to be checked.
+
+    Returns:
+        bool: True if the character is a Latin character or a number, False otherwise.
+    """
+    pattern = r'^[a-zA-Z0-9_-]$'
+    return re.match(pattern, character) is not None
```

Comment:
I just realised that it's just a single character being tested. I wonder if an in operation would be faster than a regex match. If it has to be a regex, consider pre-compiling it outside the function call.